src/dbnode/storage/index/segments.go

```diff
@@ -18,37 +18,30 @@
 // OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
 // THE SOFTWARE.
 
-package fst
+package index
 
 import (
-	"bytes"
+	"time"
 
-	"github.com/cespare/xxhash"
+	"github.com/m3db/m3/src/m3ninx/index/segment"
 )
 
-// newFSTTermsOffsetsMap returns a new fstTermsOffsetsMap with default ctor parameters.
-func newFSTTermsOffsetsMap(initialSize int) *fstTermsOffsetsMap {
-	return _fstTermsOffsetsMapAlloc(_fstTermsOffsetsMapOptions{
-		hash: func(k []byte) fstTermsOffsetsMapHash {
-			return fstTermsOffsetsMapHash(xxhash.Sum64(k))
-		},
-		equals: func(f, g []byte) bool {
-			return bytes.Equal(f, g)
-		},
-		copy:        undefinedFSTTermsOffsetsMapCopyFn,
-		finalize:    undefinedFSTTermsOffsetsMapFinalizeFn,
-		initialSize: initialSize,
-	})
+type readableSeg struct {
+	createdAt time.Time
+	segment   segment.Segment
 }
 
-var undefinedFSTTermsOffsetsMapCopyFn fstTermsOffsetsMapCopyFn = func([]byte) []byte {
-	// NB: intentionally not defined to force users of the map to not
-	// allocate extra copies.
-	panic("not implemented")
+func newReadableSeg(seg segment.Segment) *readableSeg {
+	return &readableSeg{
+		createdAt: time.Now(),
+		segment:   seg,
+	}
 }
 
-var undefinedFSTTermsOffsetsMapFinalizeFn fstTermsOffsetsMapFinalizeFn = func([]byte) {
-	// NB: intentionally not defined to force users of the map to not
-	// allocate extra copies.
-	panic("not implemented")
+func (s *readableSeg) Segment() segment.Segment {
+	return s.segment
+}
+
+func (s *readableSeg) Age() time.Duration {
+	return time.Since(s.createdAt)
```

Comment:
Should this be using a nowFn? I don't think we explicitly do `time.Now()` anywhere in the codebase

Reply:
Yeah should do, I'll update it.